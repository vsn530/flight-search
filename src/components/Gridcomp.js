import React, { useState } from 'react';
import { AgGridReact } from 'ag-grid-react';
import {columnDef} from '../data/data';
import { connect } from 'react-redux';
import getFilteredData from '../data/FilterData'
import 'ag-grid-community/dist/styles/ag-grid.css';
import 'ag-grid-community/dist/styles/ag-theme-balham.css';
import '../styles/Gridcomp.css';



const Grid = (props) => {
    
    const[columnDefs,setColumnDefs]= useState([...columnDef]);
    const gridOptions = {
        pagination:true,
        paginationPageSize :5,   
    }
    if(!props.isAuthenticated){
        return (
            <div></div>
        )
    }

    return (    

        <div id='griddiv'>
            {
                props.flightData.length > 0 ? (
                    
                    <div className='container'>
                        <p id="msg">{`Found ${props.flightData.length} flights information` }</p>
                        <div
                            className="ag-theme-balham"
                            style={{
                                height: '350px',
                                width: '100%'
                            }}
                        >
                            <AgGridReact
                                columnDefs={columnDefs}
                                rowData={props.flightData}
                                gridOptions = {gridOptions}>                                                           
                            </AgGridReact>
                        </div>
                    </div>
                ):undefined
            }
        </div>
    )
}

const mapStateToProps = (state) => {
    return {
        flightData:getFilteredData(state.flightsData,state.searchData),
        isAuthenticated:state.userData.isAuthenticated
    }
    
}

export default connect(mapStateToProps)(Grid);