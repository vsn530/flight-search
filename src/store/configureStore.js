import { createStore, combineReducers } from 'redux';
import flightReducer from '../reducers/flightsReducer';
import searchReducer from '../reducers/formDataReducer';
import userReducer from '../reducers/userReducer';


export default () => {
    const store = createStore(combineReducers({
        flightsData:flightReducer,
        searchData:searchReducer,
        userData:userReducer
    }))
    return store;
}
