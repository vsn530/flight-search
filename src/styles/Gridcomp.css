
#griddiv{
    height: 500px;
    padding: 20px;    
}

#msg{
    font-size: 20px;
    font-weight: bold;
    margin-bottom: 15px;
    margin-left: 10px;
}