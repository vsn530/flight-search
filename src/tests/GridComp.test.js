import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import SearchComp from '../components/Gridcomp';
import Gridcomp from '../components/Gridcomp';

const mockStore = configureStore([]);

describe('Tests Connected Grid Component with React-Redux ', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            flightsData:[],
            searchData: {
                sourceCity: '',
                destCity: '',
                startDate: undefined,
                returnDate: undefined
            }

        });
        store.dispatch = jest.fn();
        component = renderer.create(
            <Provider store={store}>
                <Gridcomp />
            </Provider>);

    });

    test('renders Search component correctly', () => {
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    

    // it('should dispatch an action on button click', () => {
    //     renderer.act(() => {
    //         const e = {preventDefault:jest.fn()}
    //         let button = component.root.findByType('form').props.onSubmit(e)            
    //       });
    //       expect(store.dispatch).toHaveBeenCalledTimes(1);
    // });
});