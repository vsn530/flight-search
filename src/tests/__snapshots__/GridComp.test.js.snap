// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Connected Grid Component with React-Redux  renders Search component correctly 1`] = `
<div
  id="griddiv"
/>
`;
